Formula/hadoop.rb

```diff
@@ -15,27 +15,49 @@ class Hadoop < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "92cb92589ac48035f2482eca9321cebe31dfd3873272da601e24f30cc758cb5e"
   end
 
-  depends_on "openjdk"
+  # WARNING: Check https://cwiki.apache.org/confluence/display/HADOOP/Hadoop+Java+Versions before updating JDK version
+  depends_on "openjdk@11"
 
   conflicts_with "yarn", because: "both install `yarn` binaries"
 
   def install
     rm_f Dir["bin/*.cmd", "sbin/*.cmd", "libexec/*.cmd", "etc/hadoop/*.cmd"]
     libexec.install %w[bin sbin libexec share etc]
     Dir["#{libexec}/bin/*"].each do |path|
-      (bin/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk"].opt_prefix
+      (bin/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk@11"].opt_prefix
     end
     Dir["#{libexec}/sbin/*"].each do |path|
-      (sbin/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk"].opt_prefix
+      (sbin/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk@11"].opt_prefix
     end
     Dir["#{libexec}/libexec/*.sh"].each do |path|
-      (libexec/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk"].opt_prefix
+      (libexec/File.basename(path)).write_env_script path, JAVA_HOME: Formula["openjdk@11"].opt_prefix
     end
     # Temporary fix until https://github.com/Homebrew/brew/pull/4512 is fixed
     chmod 0755, Dir["#{libexec}/*.sh"]
   end
 
   test do
     system bin/"hadoop", "fs", "-ls"
+
+    # Test if resource manager can start successfully
```

Comment:
Is there no shell script for this?

Reply:
The shell script will write log files to the libexec folder, which causes the test always to fail.

Reply:
That should probably be adjusted in the shell script either way. I don't think people would expect to find logs in the install folder. Is there no flag to specify where to log?

Reply:
I can append ```HADOOP_LOG_DIR``` to env scripts to change the log directory. But I cannot trace the pid to check if the server is alive if I use the script to start the server. The pid file contains the username in its filename, which is different for everyone and the test code would be much more complicated.